tests/solidity/test_close_short.py

```diff
@@ -0,0 +1,364 @@
+    """
+
+    contribution: float = 500_000_000
+    target_apr: float = 0.05
+    term_length: int = 365
+    alice: agent.Agent
+    bob: agent.Agent
+    celine: agent.Agent
+    hyperdrive: hyperdrive_market.Market
+
+    def setUp(self):
+        """Set up agent, pricing model, & market for the subsequent tests.
+        This function is run before each test method.
+        """
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        block_time = time.BlockTime()
+        pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.0,
+            redemption_fee_percent=0.0,
+        )
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=pricing_model,
+            market_state=market_state,
+            position_duration=time.StretchedTime(
+                days=365, time_stretch=pricing_model.calc_time_stretch(self.target_apr), normalizing_constant=365
+            ),
+            block_time=block_time,
+        )
+        _, wallet_deltas = self.hyperdrive.initialize(self.alice.wallet.address, self.contribution, 0.05)
+        self.alice.wallet.update(wallet_deltas)
+
+    def verify_close_short(
+        self,
+        example_agent: agent.Agent,
+        market_state_before: hyperdrive_market.MarketState,
+        agent_base_paid: float,
+        agent_base_proceeds: float,
+        bond_amount: float,
+        maturity_time: float,
+    ):
+        """Close a short then make sure the market state is correct"""
+        # verify that all of Bob's bonds were burned
+        self.assertFalse(
+            example_agent.wallet.shorts
+        )  # In solidity we check that the balance is zero, but here we delete the entry if it is zero
+        # verify that the bond reserves were updated according to flat+curve
+        # the adjustment should be equal to timeRemaining * bondAmount
+        time_remaining = 0
+        if maturity_time > self.hyperdrive.block_time.time:
+            time_remaining = self.hyperdrive.position_duration.normalized_time * (
+                maturity_time - self.hyperdrive.block_time.time
+            )
+        self.assertLess(  # user gets less than what they put in
+            agent_base_proceeds,
+            agent_base_paid,
+            msg="agent gets more than what they put in: agent_base_proceeds > agent_base_paid",
+        )
+        self.assertEqual(  # bond reserves
+            self.hyperdrive.market_state.bond_reserves,
+            market_state_before.bond_reserves - time_remaining * bond_amount,
+            msg="bond_reserves is wrong",
+        )
+        self.assertEqual(  # share reserves
+            self.hyperdrive.market_state.share_reserves,
+            market_state_before.share_reserves + (bond_amount - agent_base_proceeds) / market_state_before.share_price,
```

Comment:
This equation appears to be different in the solidity code?
```
        assertApproxEqAbs(
            poolInfoAfter.shareReserves,
            poolInfoBefore.shareReserves +
                (bondAmount + baseProceeds - basePaid).divDown(
                    poolInfoBefore.sharePrice
                ),
            // 10^-7
            poolInfoAfter.shareReserves.mulDown(1e11) // TODO: This error bar is too big. Analyze this.
        );
```

Reply:
It's going to match after we fix the solidity!

See here: https://github.com/element-fi/hyperdrive/blob/cf63bcbd7bc8b9e72b93eeb8567f4ab265fca6dc/test/units/hyperdrive/CloseShortTest.t.sol#L302

Alex approved the change last night.

But going over his comments we realized he doesn't want to remove the two negative interest tests. So I'm going to look at that before landing the PR. Once we do that, we'll have to circle back and re-add the negative interest tests in python as well.

Reply:
Ok sounds good